source/extensions/filters/http/jwt_authn/verifier.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include "envoy/http/async_client.h"
+
+#include "extensions/filters/http/jwt_authn/authenticator.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+class Verifier;
+typedef std::unique_ptr<Verifier> VerifierPtr;
+
+typedef std::function<AuthenticatorPtr(const std::vector<std::string>& audiences)> AuthFactory;
+
+/**
+ * Supports verification of JWTs with configured requirments.
+ */
+class Verifier {
+public:
+  virtual ~Verifier() {}
+  /**
+   * Object used to signal the caller of completion.
+   */
+  class Callbacks {
+  public:
+    virtual ~Callbacks() {}
+
+    virtual void onComplete(const ::google::jwt_verify::Status& status) PURE;
+  };
+  // Verify all tokens on headers, and signal the caller with callback.
+  virtual void verify(Http::HeaderMap& headers, Callbacks& callback) PURE;
+  // close resouces in use.
+  virtual void close() PURE;
```

Comment:
it should be called cancel()  to cancel the async verification process.

Reply:
done.